versioned_docs/version-5.x/macros-attributes/chain-extension.md

```diff
@@ -447,11 +461,94 @@ mod read_writer {
 }
 ```
 
+## Using Multiple Chain Extension
+
+It is possible to use multiple exposed chain extensions in the single environment of a smart contract.
+The declaration procedure of the chain extension stays the same.
+
+
+Suppose we want to combine two chain extension called `Psp22Extension` and `FetchRandom`, ink! provides
+a useful macro `ink::combine_extensions!` that allows to construct the structure combining 
+the aforementioned chain extensions like so:
+```rust
+ink::combine_extensions! {
+    /// This extension combines the `FetchRandom` and `Psp22Extension` extensions.
+    /// It is possible to combine any number of extensions in this way.
+    ///
+    /// This structure is an instance that is returned by the `self.env().extension()` call.
+    pub struct CombinedChainExtension {
+        /// The instance of the `Psp22Extension` chain extension.
+        ///
+        /// It provides you access to `PSP22` functionality.
+        pub psp22: Psp22Extension,
+        /// The instance of the `FetchRandom` chain extension.
+        ///
+        /// It provides you access to randomness functionality.
+        pub rand: FetchRandom,
+    }
+}
+```
+
+The combined structure is called `CombinedChainExtension`, and we can refer to it 
+when specifying the chain extension type in `Environment`:
+```rust
+type ChainExtension = CombinedChainExtension;
+```
+
+Each extension's method can be called by accessing it via the name of the field of `CombineChainExtension`:
+```rust
+self.env().extension().rand.<method_name_in_rand_ext>()
+// or
+self.env().extension().psp22.<method_name_in_psp22_ext>()
+// e.g.
+self.env().extension().psp22.total_supply()
+```
+
+:::note
+The ink! repository contains the [full example](https://github.com/paritytech/ink/tree/master/integration-tests/combined-extension) illustrating how to combine existing chain extensions 
```

Comment:
We should update this link to `ink-examples` once we've merged Sebastians PR there.